fs/background-fetcher/background_fetcher.go

```diff
@@ -0,0 +1,162 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package backgroundfetcher
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	sm "github.com/awslabs/soci-snapshotter/fs/span-manager"
+	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/containerd/containerd/log"
+	"github.com/opencontainers/go-digest"
+	"golang.org/x/time/rate"
+)
+
+type layerResolver struct {
+	*sm.SpanManager
+	layerDigest     digest.Digest
+	nextSpanFetchID soci.SpanID
+}
+
+type Option func(*BackgroundFetcher) error
+
+const (
+	// Amount of time the background fetcher will wait once a new layer comes in
+	// before (re)starting fetches.
+	defaultSilencePeriod = 5 * time.Second
+
+	// The minimum size of data to be fetched in a single fetch.
+	defaultMinFetchSize = 1 << 22 // 4 MiB
```

Comment:
How about defining mib as a constant? I cannot recall 22 bits shift is 2 MiB :)

Reply:
Ack.